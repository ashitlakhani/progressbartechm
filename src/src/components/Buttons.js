import React, { Component } from 'react';
import { Button } from 'reactstrap';

import { Constant } from '../consts/Numbers';

class Buttons extends Component {
    updateBar() {
        this.props.updateBar(this.props.elem);
    }

    render() {
        return (
            <Button
                className={`height-3rem rounded-0 mb-2
                    ${this.props.hasSelectedPb !== '' ? 'btn-outline-info' : 'btn-light'}`}
                disabled={ !this.props.hasSelectedPb }
                onClick={ () => {
                    this.updateBar();
                } }>
                {
                    (Math.sign(this.props.elem) === Constant.ONE) ? '+' : ''
                }
                { this.props.elem }
            </Button>
        );
    }
}

export default Buttons;
