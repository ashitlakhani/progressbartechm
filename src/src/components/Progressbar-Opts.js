import React from 'react';

export const ProgOptions = ({ pbId }) => (
    <option value={ pbId }>
        {pbId}
    </option>
);
