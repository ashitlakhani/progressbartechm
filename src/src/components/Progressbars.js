import React, { Component } from 'react';
import { Progress } from 'reactstrap';

export class Progressbars extends Component {
    render() {
        return (
            <div>
                {this.props.bars.map((elem, idx) => (
                    <div key={elem.pbId + idx}>
                    <div className="text-center middle-percent">
                        {elem.val}%
                    </div>
                    <Progress color={`${elem.val > this.props.limit ? 'danger' : 'info'}`}
                        className="height-3rem rounded-0"
                        value={ elem.val } />
                    </div>    
                ))}
            </div>
        );
    }
}

export default Progressbars;
