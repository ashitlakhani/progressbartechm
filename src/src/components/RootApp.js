import React from 'react';
import Container from '../containers/container';

const RootApp = () => (
    <Container/>
);

export default RootApp;
