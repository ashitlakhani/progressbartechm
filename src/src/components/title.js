import React from 'react';

const title = () => (
    <div>
        <h2>
            Progress Bar
        </h2>
        <h4>    
            Choose progressbar from the dropdown.
        </h4>
    </div>
);
export default title;