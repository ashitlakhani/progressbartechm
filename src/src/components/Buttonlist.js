import React, {Component} from 'react';
import Buttons from './Buttons';

class ButtonList extends Component {
	render() {
		return (

                <div className="text-center equalspace-flex">
                    { this.props.btns.map((elem, idx) => (
                        <Buttons elem={elem} key={elem + idx}
                            hasSelectedPb={ this.props.hasChoosen }
                            updateBar={ this.props.updateBar } />
                    )) }
                </div>

		);
	}
}

export default ButtonList;
