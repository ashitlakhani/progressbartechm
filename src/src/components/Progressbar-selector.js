import React, { Component } from 'react';
import { FormGroup, Label, Input } from 'reactstrap';

import { ProgOptions } from './Progressbar-Opts';

class ProgressbarSelector extends Component {
    changeProgressbar(e) {
        this.props.selectPbById(e);
    }

    render() {
        return (
            <FormGroup>
                <Label for="progressbAs-select0" className="text-brand">
                    Select Progressbar
                </Label>
                <Input type="select" id="progressbAs-select0"
                    name="progressbar-select" bsSize="lg"
                    className="rounded-0 normal-bdcolor-boxshadow
                        custom-select height-3rem appear-btn"
                    onChange={ (event) => {
                        this.changeProgressbar(event);
                    } }>
                    {this.props.bars.map((item, idx) => (
                        <ProgOptions key={idx} {...item} />
                    ))}
                </Input>
            </FormGroup>
        );
    }
}

export default ProgressbarSelector;
