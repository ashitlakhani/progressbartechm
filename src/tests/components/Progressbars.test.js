import React from 'react';
import { shallow } from 'enzyme';

import { Progressbars } from '../../src/components/Progressbars';
import barsArr from '../testdata/barsArr-fixtures';
import { RedBarArr } from '../testdata/singleBarItem-fixture';

describe('In Progressbars Test ', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallow(<Progressbars bars={ barsArr } limit={ 160 } />);
    });

    it('should render component correctly.', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should be changed to red when over limit.', () => {
        const wrapperWithRedBar = shallow(<Progressbars bars={ RedBarArr } limit={ 160 } />);
        const thePb = wrapperWithRedBar.find('Progress');
        expect(thePb.prop('color')).toBe('danger');
    });
});
