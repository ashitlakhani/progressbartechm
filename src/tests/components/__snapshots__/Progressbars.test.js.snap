// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`In Progressbars Test  should render component correctly. 1`] = `
<div>
  <div
    key="pb10"
  >
    <div
      className="text-center middle-percent"
    >
      10
      %
    </div>
    <Progress
      className="height-3rem rounded-0"
      color="info"
      max={100}
      tag="div"
      value={10}
    />
  </div>
  <div
    key="pb21"
  >
    <div
      className="text-center middle-percent"
    >
      81
      %
    </div>
    <Progress
      className="height-3rem rounded-0"
      color="info"
      max={100}
      tag="div"
      value={81}
    />
  </div>
  <div
    key="pb32"
  >
    <div
      className="text-center middle-percent"
    >
      42
      %
    </div>
    <Progress
      className="height-3rem rounded-0"
      color="info"
      max={100}
      tag="div"
      value={42}
    />
  </div>
</div>
`;

exports[`In components/Progressbars.js Test Progressbar:  should render component correctly. 1`] = `
<div>
  <div
    key="pb10"
  >
    <div
      className="text-center middle-percent"
    >
      10
      %
    </div>
    <Progress
      animated={true}
      className="height-3rem rounded-0"
      color="info"
      max={100}
      tag="div"
      value={10}
    />
  </div>
  <div
    key="pb21"
  >
    <div
      className="text-center middle-percent"
    >
      81
      %
    </div>
    <Progress
      animated={true}
      className="height-3rem rounded-0"
      color="info"
      max={100}
      tag="div"
      value={81}
    />
  </div>
  <div
    key="pb32"
  >
    <div
      className="text-center middle-percent"
    >
      42
      %
    </div>
    <Progress
      animated={true}
      className="height-3rem rounded-0"
      color="info"
      max={100}
      tag="div"
      value={42}
    />
  </div>
</div>
`;
